LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/MessageProcessor.cs

```diff
@@ -336,28 +342,42 @@ private async Task<byte[]> ProcessLoraMessage(LoRaMessageWrapper loraMessage)
                                 Array.Reverse(devAddrCorrect);
                                 bool requestForConfirmedResponse = false;
 
+
                                 //check if the c2d message has a mac command
                                 byte[] macbytes = null;
                                 if (c2dMsg != null)
                                 {
-                                    var macCmd = c2dMsg.Properties.Where(o => o.Key == "CidType");
-                                    if (macCmd.Count() != 0)
+                                    if (c2dMsg.Properties.TryGetValueCaseInsensitive("cidtype", out var cidTypeValue))
                                     {
-                                        MacCommandHolder macCommandHolder = new MacCommandHolder(Convert.ToByte(macCmd.First().Value));
+                                        Logger.Log(loraDeviceInfo.DevEUI, $"Cloud to device MAC command received", Logger.LoggingLevel.Info);
+                                        MacCommandHolder macCommandHolder = new MacCommandHolder(Convert.ToByte(cidTypeValue));
                                         macbytes = macCommandHolder.macCommand[0].ToBytes();
                                     }
-                                    var confirmCmd = c2dMsg.Properties.Where(o => o.Key == "Confirmed");
-                                    if (confirmCmd.Count() != 0)
+                   
+                                    if (c2dMsg.Properties.TryGetValueCaseInsensitive("confirmed", out var confirmedValue) && confirmedValue.Equals("true", StringComparison.OrdinalIgnoreCase))
                                     {
                                         requestForConfirmedResponse = true;
+                                        Logger.Log(loraDeviceInfo.DevEUI, $"Cloud to device message requesting a confirmation", Logger.LoggingLevel.Info);
+
+                                    }
+                                    if (c2dMsg.Properties.TryGetValueCaseInsensitive( "fport", out var fPortValue))
+                                    {
+                                        int fportint = int.Parse(fPortValue);

```

Comment:
We should check what we do if the value coming is not as expected:
- Let exception be raised
- Use a default and log a warning
- Ignore C2D and log a warning/error

Reply:
In general there can be many different issues like fport to big, message to big for the DR and country etc... view that we cannot control the UI we need an easy way to inform the "user" of the issue. Changing the fport number to 1 is not a good idea and probably cutting the message as we do now also not. My feeling says refuse and log as error.

Reply:
As the fport could direct this data to very different logical endpoints, I feel we should ignore the C2D and

Reply:
We can offer a CLI utility that sends the c2d where we can do some checks but we still need to manage the case that they use the UI or other tools